compiler/cpp/src/thrift/generate/t_swift_generator.cc

```diff
@@ -2262,7 +2262,7 @@ void t_swift_generator::render_const_value(ostream& out,
     out << "]";
   } else if (type->is_set()) {
     t_type* etype = ((t_set*)type)->get_elem_type();
-    out << "[";
+    out << "Set<" << type_name(etype) << ">([";
```

Comment:
We actually don't need the `<typename>` here

Reply:
Is there any downside to specifying a typename in the set?

Reply:
It’s a Swift pattern to let the compiler determine the type. Definitely in OM, and I believe in Swift in general.

To answer your question, I think the only advantage would be to more easily change between types. But really the main advantage here is that it’s the syntax pattern we follow in OM

Reply:
I could see the value in keeping the type name if we were to do this:

```
let days = Set<DayOfWeek>([.monday, .friday])
```

But I think that would be more code since we can’t reuse that const value function

Reply:
I've removed the type declaration from the const declaration.

`public var weekend = Set([DayOfTheWeek.saturday, DayOfTheWeek.sunday])`

The type is still present in the type alias when constant values are missing:

`public typealias DaysOfTheWeekNonTelemetry = Set<DayOfTheWeek>`